### PR TITLE
[query] Rewrite ExportEntriesByCol to stage locally

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/src/main/scala/is/hail/methods/MatrixExportEntriesByCol.scala [hunk 1]
```diff
@@ -69,8 +69,14 @@ case class MatrixExportEntriesByCol(parallelism: Int, path: String, bgzip: Boole
 
         val partFolder = partFileBase + partFile(d, i, TaskContext.get())
 
-        val fileHandles = Array.tabulate(endIdx - startIdx) { j =>
-          new OutputStreamWriter(bcFS.value.create(partFolder + "/" + j.toString + extension), "UTF-8")
+        val filePaths = Array.tabulate(endIdx - startIdx) { j =>
+          val finalPath = partFolder + "/" + j.toString + extension
+          val tempPath = bcFS.value.getTemporaryFile("file:///tmp", suffix=Some(extension))
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/methods/MatrixExportEntriesByCol.scala:74]: this is going to conflict with cotton's change I believe, will need to be extension=

Comment (akotlar) [reply hail/src/main/scala/is/hail/methods/MatrixExportEntriesByCol.scala:74]: Actually, now it will be createTmpPath, which lives on ExecuteContext

Comment (akotlar) [reply hail/src/main/scala/is/hail/methods/MatrixExportEntriesByCol.scala:74]: https://github.com/hail-is/hail/pull/8581/files#diff-8a2e24e243a3e3f841d27ef41a52a681R55

Comment (tpoterba) [reply hail/src/main/scala/is/hail/methods/MatrixExportEntriesByCol.scala:74]: will tag this as WIP, will be easier for me to rebase than him.

Comment (tpoterba) [reply hail/src/main/scala/is/hail/methods/MatrixExportEntriesByCol.scala:74]: thanks for the heads up

**Pull request merged**